src/WebpackTools/MagentoResolver.js

```diff
@@ -0,0 +1,15 @@
+const optionsValidator = require('../util/options-validator');
+const validateConfig = optionsValidator('MagentoResolver', {
+    'paths.root': 'string'
+});
+module.exports = {
+    validateConfig,
+    async configure(options) {
+        validateConfig('.configure()', options);
+        return {
+            modules: [options.paths.root, 'node_modules'],
+            mainFiles: ['index'],
+            extensions: ['.js']
```

Comment:
This is overriding webpack's default `.json` entry, which is going to surprise anyone who is used to JSON imports working with node/webpack OOTB